src/dlgProfilePreferences.cpp

```diff
@@ -2093,6 +2119,78 @@ void dlgProfilePreferences::slot_logFileNameFormatChange(const int index)
     label_logFileNameExtension->setVisible(isShown);
 }
 
+void dlgProfilePreferences::slot_configDirOptionChange(const int index)
+{
+    Q_UNUSED(index);
+
+    // Show options to set the config directory if 'Select a
+    // directory...' is selected
+    bool isShown = index == 1;
+    label_setConfigDir->setVisible(isShown);
+    lineEdit_configDir->setVisible(isShown);
+    toolButton_setConfigDir->setVisible(isShown);
+
+    switch (index) {
+    case 0:
+        // Clear mConfigDirPath if 'Default' is selected
+        mConfigDirPath.clear();
+        lineEdit_configDir->setText(QDir::homePath() + "/.config/mudlet");
+        break;
+    case 1:
+        // Set mConfigDirPath to the value of lineEdit_configDir if
+        // 'Select a directory' is selected
+        mConfigDirPath = lineEdit_configDir->text();
+        break;
+    case 2:
+        // Set mConfigDirPath to a relative path if 'Relative to the
+        // application file' is selected
+        mConfigDirPath = ".";
+        lineEdit_configDir->setText(QCoreApplication::applicationDirPath() + "/config");
```

Comment:
This one is not in `getMudletPath(...)` currently - but it would be valid to add another `case` to handle it.